api/v1beta1/etcdadmcluster_webhook_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestValidateCommon(t *testing.T) {
+	cases := map[string]struct {
+		in        *EtcdadmCluster
+		expectErr bool
```

Comment:
I suggest you expect particular error messages
It helps avoid situations where the method returns an error you don't expect but you test can't catch it